cmake/Dependencies.cmake

```diff
@@ -1260,6 +1260,7 @@ if(USE_ROCM)
     list(APPEND HIP_CXX_FLAGS -fPIC)
     list(APPEND HIP_CXX_FLAGS -D__HIP_PLATFORM_HCC__=1)
     list(APPEND HIP_CXX_FLAGS -DCUDA_HAS_FP16=1)
+    list(APPEND HIP_CXX_FLAGS -DUSE_ROCM)
     list(APPEND HIP_CXX_FLAGS -D__HIP_NO_HALF_OPERATORS__=1)
```

Comment:
@dllehr-amd Can you please mention here why we needed to define USE_ROCM here as well, just to capture it for future reference?

Reply:
So we needed the USE_ROCM flag available in the c10 code base.  Without this flag we end up with getting compilation errors trying to use `USE_ROCM` in c10/cuda/CUDACachingAllocator.cpp

I tried a few areas to get around this, if there's a better alternative to this I'm up for trying

Reply:
Seems odd.  USE_ROCM is sprinkled throughout the codebase as a replacement for `__HIP_PLATFORM_HCC__`.  It must be getting set as a preprocessor symbol somewhere.

Reply:
Yeah I rebuilt with that flag off, and I'm not getting compile errors.  It may have been a weird setting in my build, or the problem resolved itself since I last tried a few months ago.  I'll take this out